prompt2model/model_trainer/base.py

```diff
@@ -25,6 +25,28 @@ def __init__(self, pretrained_model_name: str):
         )
         self.wandb = None
 
+    # we can also have another method called register_trainer
+    # or something like this
+    @property
+    def trainer(self) -> transformers.Trainer:
+        """Access the hugging face trainer class.
+
+        Returns:
+            transformers.Trainer: The hugging face trainer class
+        """
+        # Let's assume we are doing Seq2Seq training here
+        # Also assuming the model has encoder now, just for the sake of not changing the
+        # constructor's defination
+
+        def model_init():
+            return self.model
+
+        return transformers.Seq2SeqTrainer(
+            model=self.model,
```

Comment:
I think this is ok for now, but I'm wondering if we can instead use the actual trainer used by the model. We might need to add a `self.hf_trainer` member variable to support that.

Reply:
If we stick with the current formulation, can we create an issue to record this restriction that we're making in the parameter selector?

Reply:
Well, yes we have an existing problem where we have too many trainers but with different meanings, like `self.trainer` or `trainer` from hugging face and `trainer` from `BaseTrainer` class. So we can have a `self.hf_trainer` can make this more cleaner. Can you bit elaborate how you want to see that in action?

Reply:
Since you refactored the code, it seems that this can all be removed? This property method doesn't seem to be used anywhere.

Reply:
Yes, we can remove this for now.